frontend/packages/helm-plugin/src/utils/helm-utils.ts

```diff
@@ -100,7 +99,28 @@ export const getChartRepositoryTitle = (
   chartRepoName: string,
 ) => {
   const chartRepository = chartRepositories?.find((repo) => repo.metadata.name === chartRepoName);
-  return chartRepository?.spec?.name || toTitleCase(chartRepoName);
+  if (chartRepository) {
+    return chartRepository?.spec?.name;
+  }
```

Comment:
The display name is optional, so this code should fallback to the chartRepoName when it's null, undefined or empty.

```suggestion
  if (chartRepository?.spec?.name) {
    return chartRepository.spec.name;
  }
```